packages/syft/src/syft/service/project/distributed_project.py

```diff
@@ -129,6 +129,11 @@ def _populate_auto_generated_fields(self) -> Self:
         self.members = self._get_members_from_clients()
         return self
 
+    @model_validator(mode="after")
```

Comment:
There's already a field validator for the `code` attribute. Could we move this logic there?
Additionally, it would be great to add a few conditionals: one to ensure this logic only applies to `SubmitUserCode` (since the type is `UserCode | SubmitUserCode`), and another to check that the ID is not already assigned to prevent overwriting it.